fdbserver/ClusterController.actor.cpp

```diff
@@ -407,98 +408,44 @@ class ClusterControllerData {
+				requiredFitness = fitness;
+				requiredUsed = used;
 			}
 
+			if (std::get<2>(workerIter->first)) {
+				requiredDegraded = true;
+			}
+			if (std::get<3>(workerIter->first)) {
+				requiredInCCDC = true;
+			}
 			for (auto& worker : workerIter->second) {
 				logServerMap->add(worker.interf.locality, &worker);
 			}
-
-			if (logServerSet->size() < (std::get<2>(workerIter->first) ? required : desired)) {
-			} else if (logServerSet->size() == required || logServerSet->size() <= desired) {
-				if (logServerSet->validate(policy)) {
-					for (auto& object : logServerMap->getObjects()) {
-						results.push_back(*object);
-					}
-					bCompleted = true;
-					break;
-				}
-				TraceEvent(SevWarn, "GWFTADNotAcceptable", id)
-				    .detail("DcIds", dcList)
-				    .detail("Fitness", fitness)
-				    .detail("Processes", logServerSet->size())
-				    .detail("Required", required)
-				    .detail("TLogPolicy", policy->info())
-				    .detail("DesiredLogs", desired)
-				    .detail("Used", used)
-				    .detail("AddingDegraded", addingDegraded);
-			}
-			// Try to select the desired size, if larger
-			else {
-				std::vector<LocalityEntry> bestSet;
-				std::vector<LocalityData> tLocalities;
-
-				// Try to find the best team of servers to fulfill the policy
-				if (findBestPolicySet(bestSet,
-				                      logServerSet,
-				                      policy,
-				                      desired,
-				                      SERVER_KNOBS->POLICY_RATING_TESTS,
-				                      SERVER_KNOBS->POLICY_GENERATIONS)) {
-					results.reserve(results.size() + bestSet.size());
-					for (auto& entry : bestSet) {
-						auto object = logServerMap->getObject(entry);
-						ASSERT(object);
-						results.push_back(*object);
-						tLocalities.push_back(object->interf.locality);
-					}
-					TraceEvent("GWFTADBestResults", id)
-					    .detail("DcIds", dcList)
-					    .detail("Fitness", fitness)
-					    .detail("Used", used)
-					    .detail("Processes", logServerSet->size())
-					    .detail("BestCount", bestSet.size())
-					    .detail("BestZones", ::describeZones(tLocalities))
-					    .detail("BestDataHalls", ::describeDataHalls(tLocalities))
-					    .detail("TLogPolicy", policy->info())
-					    .detail("TotalResults", results.size())
-					    .detail("DesiredLogs", desired)
-					    .detail("AddingDegraded", addingDegraded);
-					bCompleted = true;
-					break;
-				}
-				TraceEvent(SevWarn, "GWFTADNoBest", id)
-				    .detail("DcIds", dcList)
-				    .detail("Fitness", fitness)
-				    .detail("Used", used)
-				    .detail("Processes", logServerSet->size())
-				    .detail("Required", required)
-				    .detail("TLogPolicy", policy->info())
-				    .detail("DesiredLogs", desired)
-				    .detail("AddingDegraded", addingDegraded);
-			}
 		}
 
-		// If policy cannot be satisfied
-		if (!bCompleted) {
+		if (!bCompleted && !(logServerSet->size() >= required && logServerSet->validate(policy))) {
```

Comment:
The second condition appears always to be false when bCompleted is false (from line 430 above). If so, please consider removing it.